<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CBEA">
<title>CBEA: Competitive Balances for Taxonomic Enrichment Analysis • CBEA</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CBEA: Competitive Balances for Taxonomic Enrichment Analysis">
<meta property="og:description" content="CBEA">
<meta property="og:image" content="https://qpmnguyen.github.io/CBEA/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CBEA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/basic_usage.html">CBEA: Competitive Balances for Taxonomic Enrichment Analysis</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/qpmnguyen/CBEA/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="basic_usage_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>CBEA: Competitive Balances for Taxonomic Enrichment Analysis</h1>
                        <h4 data-toc-skip class="author">Quang P.Nguyen</h4>
            <address class="author_afil">
      Department of Epidemiology at Dartmouth College<br><a class="author_email" href="mailto:#"></a><a href="mailto:quangpmnguyen@gmail.com" class="email">quangpmnguyen@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Anne G. Hoen</h4>
            <address class="author_afil">
      Department of Epidemiology at Dartmouth College<br><a class="author_email" href="mailto:#"></a><a href="mailto:Anne.G.Hoen@Dartmouth.edu" class="email">Anne.G.Hoen@Dartmouth.edu</a>
      </address>
                              <h4 data-toc-skip class="author">H. Robert Frost</h4>
            <address class="author_afil">
      Department of Biomedical Data Science at Dartmouth College<br><a class="author_email" href="mailto:#"></a><a href="mailto:Hildreth.R.Frost@Dartmouth.edu" class="email">Hildreth.R.Frost@Dartmouth.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">3 March 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/qpmnguyen/CBEA/blob/HEAD/vignettes/basic_usage.Rmd" class="external-link"><code>vignettes/basic_usage.Rmd</code></a></small>
      <div class="d-none name"><code>basic_usage.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This package implements the CBEA approach for performing set-based enrichment analysis for microbiome relative abundance data. A preprint of the package can be found <a href="https://www.biorxiv.org/content/10.1101/2021.09.07.459294v1.full" class="external-link">on bioXriv</a>. In summary, CBEA (Competitive Balances for taxonomic Enrichment Analysis) provides an estimate of the activity of a set by transforming an input taxa-by-sample data matrix into a corresponding set-by-sample data matrix. The resulting output can be used for additional downstream analyses such as differential abundance, classification, clustering, etc. using set-based features instead of the original units.</p>
<p>The transformation that CBEA applies is based on the isometric log ratio transformation:<br><span class="math display">\[
CBEA_{i,\mathbb{S}} = \sqrt{\frac{|\mathbb{S}||\mathbb{S_c}|}{|\mathbb{S}| + |\mathbb{S_c}|}} \ln \frac{g(X_{i,j | j\in \mathbb{S}})}{g(X_{i,j | j \notin \mathbb{S}})}
\]</span> Where <span class="math inline">\(\mathbb{S}\)</span> is the set of interest, <span class="math inline">\(\mathbb{S}_C\)</span> is it’s complement, <span class="math inline">\(g()\)</span> is the geometric mean operation, and <span class="math inline">\(X\)</span> is the original data matrix where <span class="math inline">\(i\)</span> is the index representing samples and <span class="math inline">\(j\)</span> is the index representing variables (or taxa).</p>
<p>The inference procedure is performed through estimating the null distribution of the test statistic. This can be done either via permutations or a parametric fit of a distributional form on the permuted scores. Users can also adjust for variance inflation due to inter-taxa correlation. Please refer to the main manuscript for any additional details.</p>
</div>
<div class="section level2">
<h2 id="usage-guide">Usage guide<a class="anchor" aria-label="anchor" href="#usage-guide"></a>
</h2>
<div class="section level3">
<h3 id="install-cbea">Install <code>CBEA</code><a class="anchor" aria-label="anchor" href="#install-cbea"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.14/CBEA" class="external-link">CBEA</a></em> is an <code>R</code> package available via the <a href="http://bioconductor.org" class="external-link">Bioconductor</a> repository for packages. It requires installing the <code>R</code> open source statistical programming language, which can be accessed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a>. After which you can install <em><a href="https://bioconductor.org/packages/3.14/CBEA" class="external-link">CBEA</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
  <span class="op">}</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"CBEA"</span><span class="op">)</span>

<span class="co">## Check that you have a valid Bioconductor installation</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>If there are any issues with the installation procedure or package features, the best place would be to file an issue at the GitHub <a href="https://github.com/qpmnguyen/CBEA" class="external-link">repository</a>. For any additional support you can use the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a> and use the <code>CBEA</code> tag and check <a href="https://support.bioconductor.org/t/CBEA/" class="external-link">the older posts</a>. Please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/qpmnguyen/CBEA" class="external-link">"CBEA"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BiocSet</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1020</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="loading-sample-data">Loading sample data<a class="anchor" aria-label="anchor" href="#loading-sample-data"></a>
</h3>
<p>First, we can load some pre-packaged data sets contained in <em><a href="https://bioconductor.org/packages/3.14/CBEA" class="external-link">CBEA</a></em>. Here we’re loading the data from the Human Microbiome Project (HMP) in <code>TreeSummarizedExperiment</code> data container from the <em><a href="https://bioconductor.org/packages/3.14/TreeSummarizedExperiment" class="external-link">TreeSummarizedExperiment</a></em>. This package does not support <code>phyloseq</code> from the <em><a href="https://bioconductor.org/packages/3.14/phyloseq" class="external-link">phyloseq</a></em> package but users can leverage the <em><a href="https://bioconductor.org/packages/3.14/mia" class="external-link">mia</a></em> package to convert between the data types.</p>
<p>In addition, users can also input raw matrices or data frames, however those require additional arguments. The <code>taxa_are_rows</code> argument requires users specify whether the data frame/matrix has taxa abundances as rows (or as columns). The <code>id_col</code> argument requires users to specify (for data frames only) a vector of names of row metadata that will be excluded from the analysis.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"hmp_gingival"</span><span class="op">)</span>
<span class="va">abun</span> <span class="op">&lt;-</span> <span class="va">hmp_gingival</span><span class="op">$</span><span class="va">data</span>
<span class="va">metab_sets</span> <span class="op">&lt;-</span> <span class="va">hmp_gingival</span><span class="op">$</span><span class="va">set</span>
<span class="va">abun</span> <span class="co"># this is a TreeSummarizedExperiment object </span>
<span class="co">#&gt; class: TreeSummarizedExperiment </span>
<span class="co">#&gt; dim: 5378 369 </span>
<span class="co">#&gt; metadata(4): experimentData phylogeneticTree experimentData</span>
<span class="co">#&gt;   phylogeneticTree</span>
<span class="co">#&gt; assays(1): 16SrRNA</span>
<span class="co">#&gt; rownames(5378): OTU_97.10005 OTU_97.10006 ... OTU_97.9991 OTU_97.9995</span>
<span class="co">#&gt; rowData names(7): CONSENSUS_LINEAGE SUPERKINGDOM ... FAMILY GENUS</span>
<span class="co">#&gt; colnames(369): 700014427 700014521 ... 700111587 700111758</span>
<span class="co">#&gt; colData names(7): RSID VISITNO ... HMP_BODY_SUBSITE SRS_SAMPLE_ID</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; rowLinks: NULL</span>
<span class="co">#&gt; rowTree: NULL</span>
<span class="co">#&gt; colLinks: NULL</span>
<span class="co">#&gt; colTree: NULL</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="input-sets">Input sets<a class="anchor" aria-label="anchor" href="#input-sets"></a>
</h3>
<p>CBEA accepts any type of sets, as long as it is in the <code>BiocSet</code> format where the elements in the sets can be matched to taxa names in the data set. The main function will check if these names match.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">metab_sets</span>
<span class="co">#&gt; class: BiocSet</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; es_element():</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 4,828 × 1</span></span>
<span class="co">#&gt;   element     </span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> OTU_97.10005</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> OTU_97.10053</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> OTU_97.10090</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 4,825 more rows</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; es_set():</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 1</span></span>
<span class="co">#&gt;   set        </span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> F Anaerobic</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Aerobic    </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Anaerobic  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; es_elementset() &lt;active&gt;:</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 4,828 × 2</span></span>
<span class="co">#&gt;   element      set        </span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> OTU_97.10005 F Anaerobic</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> OTU_97.10053 F Anaerobic</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> OTU_97.10090 F Anaerobic</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 4,825 more rows</span></span></code></pre></div>
<p>For more information on <code>BiocSet</code>, please refer to the documentation from <em><a href="https://bioconductor.org/packages/3.14/BiocSet" class="external-link">BiocSet</a></em>. However, simply speaking, <code>BiocSet</code> acts similar to a list of three data frames and can be used in conjunction with <em><a href="https://CRAN.R-project.org/package=dplyr" class="external-link">dplyr</a></em>/<em><a href="https://CRAN.R-project.org/package=tidyr" class="external-link">tidyr</a></em>.</p>
</div>
<div class="section level3">
<h3 id="applying-cbea">Applying CBEA<a class="anchor" aria-label="anchor" href="#applying-cbea"></a>
</h3>
<p>After specifying the inputs, <code>cbea</code> is the main function to apply the method. If there are zeros in the abundance data, the <code>cbea</code> will add a pseudocount to avoid issues with the log-ratio transformation (but will throw a warning). If a different zero-handling approach is desired, users should pre-process the abundance data with the appropriate method. For parametric fits, <code>cbea</code> relies on the <em><a href="https://CRAN.R-project.org/package=fitdistrplus" class="external-link">fitdistrplus</a></em> and <code>*[mixtools](https://CRAN.R-project.org/package=mixtools)*</code> packages to estimate the parameters of the null. Specific arguments to control this fitting procedure can be provided as a named list in the <code>control</code> argument.<br>
Applying <code>cbea</code> is one command:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cbea.html">cbea</a></span><span class="op">(</span><span class="va">abun</span>, set <span class="op">=</span> <span class="va">metab_sets</span>, abund_values <span class="op">=</span> <span class="st">"16SrRNA"</span>,
              output <span class="op">=</span> <span class="st">"cdf"</span>, distr <span class="op">=</span> <span class="st">"mnorm"</span>, adj <span class="op">=</span> <span class="cn">TRUE</span>, thresh <span class="op">=</span> <span class="fl">0.05</span>, n_perm <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt; Warning in .local(obj, set, output, distr, adj, n_perm, parametric, thresh, : Taxonomic count table contains zeros,</span>
<span class="co">#&gt;             which would invalidate the log-ratio transform.</span>
<span class="co">#&gt;             Adding a pseudocount of 1e-5...</span>
<span class="co">#&gt; number of iterations= 644 </span>
<span class="co">#&gt; number of iterations= 1253 </span>
<span class="co">#&gt; number of iterations= 468 </span>
<span class="co">#&gt; number of iterations= 847 </span>
<span class="co">#&gt; number of iterations= 612 </span>
<span class="co">#&gt; number of iterations= 105</span>
<span class="va">results</span>
<span class="co">#&gt; CBEA output of class 'CBEAout' with 369 samples and 3 sets </span>
<span class="co">#&gt;  Fit type: Parametric with 2-component Gaussian Mixture Distribution </span>
<span class="co">#&gt;  Number of permutations: 10 </span>
<span class="co">#&gt;  Output type: CDF values (cdf)</span></code></pre></div>
<p>Some important arguments to control the behaviour of CBEA.</p>
<ul>
<li>
<code>output</code>: This controls what type of output is being returned. CBEA usually estimates a parametric null and users can specify what they want in return. If users want to perform downstream analysis with set-level features, they can return CDF values or z-scores of each raw score computed against that distribution (options <code>cdf</code> or <code>zscore</code>). Alternatively, users can just return the raw scores themselves (no distribution fitting will be performed) using <code>raw</code> as the option. Users can also use this distribution to estimate unadjusted p-values (option <code>pval</code>) to see whether a set is enriched at each sample. These unadjusted p-values can be converted based on a threshold (based on <code>thresh</code> which is default to be set at 0.05) into a dummy variable indicating enrichment (option <code>sig</code>). <strong>Note</strong>: CDF values and Z-scores are not available for non-parametric null estimations.<br>
</li>
<li>
<code>parametric</code>: This is a logical argument to specify whether a the null distribution will be specified via parametric fit or via non-parametric permutation testing. If <code>parametric</code> is <code>TRUE</code>, users need to specify <code>distr</code> and <code>adj</code>. If <code>parametric</code> is <code>FALSE</code>, users need to increase <code>n_perm</code>.<br>
</li>
<li>
<code>distr</code>: The form of the distribution if parametric fit is desired. As of now only supports <code>norm</code>, <code>mnorm</code>.<br>
</li>
<li>
<code>adj</code>: Whether the distribution should be adjusted for variance inflation. This procedure is done by combining the mean estimate from scores computed from permuted data set and the variance estimate from raw scores (computed on the unpermuted data set).</li>
</ul>
</div>
<div class="section level3">
<h3 id="model-output">Model output<a class="anchor" aria-label="anchor" href="#model-output"></a>
</h3>
<p>The output object is of class <code>CBEAout</code>, which is an S3 object. The underlying data structure is a list of lists, where the outer lists represent different aspects of the output. For example <code>R</code> represent the final scores while <code>diagnostic</code> represent certain goodness-of-fit statistics.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span>
<span class="co">#&gt; [1] "R"              "parameters"     "diagnostic"     "fit_comparison"</span>
<span class="co">#&gt; [5] "call_param"</span></code></pre></div>
<p>Within each aspect, there is a list of size equivalent to the total number of sets evaluated. For example, the <code>results</code> object is of size 3 representing the evaluated sets.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">R</span><span class="op">)</span>
<span class="co">#&gt; List of 3</span>
<span class="co">#&gt;  $ Aerobic    : num [1:369] 0.9694 0.0286 0.9953 0.8842 0.5617 ...</span>
<span class="co">#&gt;  $ Anaerobic  : num [1:369] 0.000461 0.58112 0.203075 0.034167 0.088761 ...</span>
<span class="co">#&gt;  $ F Anaerobic: num [1:369] 0.997 0.89 0.303 0.96 0.863 ...</span></code></pre></div>
<p>Users can use <code>tidy</code> and <code>glance</code> following the <em><a href="https://CRAN.R-project.org/package=broom" class="external-link">broom</a></em> to process <code>CBEAout</code> into nice objects. The <code>tidy</code> function returns a <code>tibble</code> of scores (samples by set). The <code>glance</code> function returns some diagnostics. There are two options for the <code>glance</code> function: <code>fit_comparison</code> allows users to compare the l-moments of the data, the permuted data, and the final fitted distribution; <code>fit_diagnostic</code> shows goodness-of-fit statistics of the distribution fitting procedure itself, with log-likelihoods and Anderson-Darling (column “ad”) statistics.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://generics.r-lib.org/reference/tidy.html" class="external-link">tidy</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 369 × 4</span></span>
<span class="co">#&gt;    sample_ids Aerobic Anaerobic F_Anaerobic</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 700014427   0.969   0.000<span style="text-decoration: underline;">461</span>      0.997 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 700014521   0.028<span style="text-decoration: underline;">6</span>  0.581         0.890 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 700014603   0.995   0.203         0.303 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 700014749   0.884   0.034<span style="text-decoration: underline;">2</span>        0.960 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 700014791   0.562   0.088<span style="text-decoration: underline;">8</span>        0.863 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 700014917   0.720   0.134         0.898 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 700014989   0.024<span style="text-decoration: underline;">9</span>  0.992         0.046<span style="text-decoration: underline;">2</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 700015076   0.685   0.069<span style="text-decoration: underline;">8</span>        0.911 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 700015149   0.402   0.000<span style="text-decoration: underline;">344</span>      1.00  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 700015215   0.097<span style="text-decoration: underline;">6</span>  0.729         0.605 </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 359 more rows</span></span>
<span class="fu"><a href="https://generics.r-lib.org/reference/glance.html" class="external-link">glance</a></span><span class="op">(</span><span class="va">results</span>, <span class="st">"fit_comparison"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 9 × 7</span></span>
<span class="co">#&gt;   set_ids     final_param  distr l_location l_scale l_skewness l_kurtosis</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;named list&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Aerobic     <span style="color: #949494;">&lt;dbl [6]&gt;</span>    data      6.98     15.2   0.030<span style="text-decoration: underline;">0</span>        0.114 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Aerobic     <span style="color: #949494;">&lt;dbl [6]&gt;</span>    perm     -<span style="color: #BB0000;">0.055</span><span style="color: #BB0000; text-decoration: underline;">7</span>    2.58 -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">012</span><span style="color: #BB0000;">3</span>     0.131 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Aerobic     <span style="color: #949494;">&lt;dbl [6]&gt;</span>    fit       0.076<span style="text-decoration: underline;">8</span>   13.0  -<span style="color: #BB0000;">0.004</span><span style="color: #BB0000; text-decoration: underline;">01</span>       0.127 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Anaerobic   <span style="color: #949494;">&lt;dbl [6]&gt;</span>    data     -<span style="color: #BB0000;">6.42</span>     20.0   0.036<span style="text-decoration: underline;">0</span>        0.091<span style="text-decoration: underline;">7</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Anaerobic   <span style="color: #949494;">&lt;dbl [6]&gt;</span>    perm      0.758     2.54  0.014<span style="text-decoration: underline;">6</span>        0.122 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Anaerobic   <span style="color: #949494;">&lt;dbl [6]&gt;</span>    fit       0.986    15.0   0.010<span style="text-decoration: underline;">4</span>        0.132 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">7</span> F_Anaerobic <span style="color: #949494;">&lt;dbl [6]&gt;</span>    data      5.88     16.5   0.048<span style="text-decoration: underline;">1</span>        0.095<span style="text-decoration: underline;">2</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">8</span> F_Anaerobic <span style="color: #949494;">&lt;dbl [6]&gt;</span>    perm     -<span style="color: #BB0000;">0.465</span>     2.71  0.000<span style="text-decoration: underline;">482</span>      0.125 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">9</span> F_Anaerobic <span style="color: #949494;">&lt;dbl [6]&gt;</span>    fit      -<span style="color: #BB0000;">0.591</span>    15.1   0.000<span style="text-decoration: underline;">309</span>      0.123</span>
<span class="fu"><a href="https://generics.r-lib.org/reference/glance.html" class="external-link">glance</a></span><span class="op">(</span><span class="va">results</span>, <span class="st">"fit_diagnostic"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 5</span></span>
<span class="co">#&gt;   set_ids     final_param   loglik    ad type          </span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;named list&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Aerobic     <span style="color: #949494;">&lt;dbl [6]&gt;</span>    -<span style="color: #BB0000; text-decoration: underline;">10</span><span style="color: #BB0000;">853.</span> 0.229 permuted_distr</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Aerobic     <span style="color: #949494;">&lt;dbl [6]&gt;</span>     -<span style="color: #BB0000; text-decoration: underline;">1</span><span style="color: #BB0000;">735.</span> 0.141 unperm_distr  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Anaerobic   <span style="color: #949494;">&lt;dbl [6]&gt;</span>    -<span style="color: #BB0000; text-decoration: underline;">10</span><span style="color: #BB0000;">791.</span> 0.327 permuted_distr</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Anaerobic   <span style="color: #949494;">&lt;dbl [6]&gt;</span>     -<span style="color: #BB0000; text-decoration: underline;">1</span><span style="color: #BB0000;">832.</span> 0.185 unperm_distr  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> F_Anaerobic <span style="color: #949494;">&lt;dbl [6]&gt;</span>    -<span style="color: #BB0000; text-decoration: underline;">11</span><span style="color: #BB0000;">036.</span> 0.321 permuted_distr</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">6</span> F_Anaerobic <span style="color: #949494;">&lt;dbl [6]&gt;</span>     -<span style="color: #BB0000; text-decoration: underline;">1</span><span style="color: #BB0000;">762.</span> 0.277 unperm_distr</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="parallel-computing">Parallel computing<a class="anchor" aria-label="anchor" href="#parallel-computing"></a>
</h3>
<p><code>CBEA</code> has in-built capacity to perform calculations paralelled across the total number of sets. The engine for parallelization is <em><a href="https://bioconductor.org/packages/3.14/BiocParallel" class="external-link">BiocParallel</a></em>. If <code>NULL</code>, <code>SerialParam</code> backend will be used.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">registered</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; $SerialParam</span>
<span class="co">#&gt; class: SerialParam</span>
<span class="co">#&gt;   bpisup: FALSE; bpnworkers: 1; bptasks: 0; bpjobname: BPJOB</span>
<span class="co">#&gt;   bplog: FALSE; bpthreshold: INFO; bpstopOnError: TRUE</span>
<span class="co">#&gt;   bpRNGseed: ; bptimeout: 2592000; bpprogressbar: FALSE</span>
<span class="co">#&gt;   bpexportglobals: TRUE; bpforceGC: FALSE</span>
<span class="co">#&gt;   bplogdir: NA</span>
<span class="co">#&gt;   bpresultdir: NA</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cbea.html">cbea</a></span><span class="op">(</span><span class="va">abun</span>, set <span class="op">=</span> <span class="va">metab_sets</span>, abund_values <span class="op">=</span> <span class="st">"16SrRNA"</span>,
     output <span class="op">=</span> <span class="st">"cdf"</span>, distr <span class="op">=</span> <span class="st">"mnorm"</span>, adj <span class="op">=</span> <span class="cn">TRUE</span>, thresh <span class="op">=</span> <span class="fl">0.05</span>, n_perm <span class="op">=</span> <span class="fl">10</span>, 
     parallel_backend <span class="op">=</span> <span class="fu">MulticoreParam</span><span class="op">(</span>workers <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="citing-cbea">Citing <code>CBEA</code><a class="anchor" aria-label="anchor" href="#citing-cbea"></a>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.14/CBEA" class="external-link">CBEA</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Citation info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"CBEA"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Nguyen Q (2022). _CBEA: R package for performing CBEA approach_. doi:</span>
<span class="co">#&gt; 10.18129/B9.bioc.CBEA (URL: https://doi.org/10.18129/B9.bioc.CBEA),</span>
<span class="co">#&gt; https://github.com/qpmnguyen/CBEA - R package version 0.99.3, &lt;URL:</span>
<span class="co">#&gt; http://www.bioconductor.org/packages/CBEA&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Nguyen Q (2022). "CBEA: Competitive balances for taxonomic enrichment</span>
<span class="co">#&gt; analysis." _bioRxiv_. doi: 10.1101/TODO (URL:</span>
<span class="co">#&gt; https://doi.org/10.1101/TODO), &lt;URL:</span>
<span class="co">#&gt; https://www.biorxiv.org/content/10.1101/TODO&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span>
<span class="co">#&gt; bibtex=TRUE)', 'toBibtex(.)', or set</span>
<span class="co">#&gt; 'options(citation.bibtex.max=999)'.</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.14/CBEA" class="external-link">CBEA</a></em> package (Nguyen, 2022) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2021)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.14/BiocStyle" class="external-link">BiocStyle</a></em> (Oleś, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em> (McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em> (Allaire, Xie, McPherson, Luraschi, Ushey, Atkins, Wickham, Cheng, Chang, and Iannone, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=broom" class="external-link">broom</a></em> (Robinson, Hayes, and Couch, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em> (Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em> (Wickham, 2011)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=mixtools" class="external-link">mixtools</a></em> (Benaglia, Chauveau, Hunter, and Young, 2009)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=fitdistrplus" class="external-link">fitdistrplus</a></em> (Delignette-Muller and Dutang, 2015)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=tidyverse" class="external-link">tidyverse</a></em> (Wickham, Averick, Bryan, Chang, McGowan, François, Grolemund, Hayes, Henry, Hester, Kuhn, Pedersen, Miller, Bache, Müller, Ooms, Robinson, Seidel, Spinu, Takahashi, Vaughan, Wilke, Woo, and Yutani, 2019)</li>
<li><em><a href="https://bioconductor.org/packages/3.14/BiocSet" class="external-link">BiocSet</a></em></li>
<li>
<em><a href="https://bioconductor.org/packages/3.14/phyloseq" class="external-link">phyloseq</a></em> (McMurdie and Holmes, 2013)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.14/BiocParallel" class="external-link">BiocParallel</a></em> (Morgan, Wang, Obenchain, Lang, Thompson, and Turaga, 2021)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.14/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Create the vignette</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"basic_usage.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Extract the R code</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"basic_usage.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span class="co">#&gt; [1] "2022-03-03 18:58:24 UTC"</span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span class="co">#&gt; Time difference of 20.336 secs</span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span>
<span class="co">#&gt;  version  R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt;  os       Ubuntu 20.04.4 LTS</span>
<span class="co">#&gt;  system   x86_64, linux-gnu</span>
<span class="co">#&gt;  ui       X11</span>
<span class="co">#&gt;  language en</span>
<span class="co">#&gt;  collate  C.UTF-8</span>
<span class="co">#&gt;  ctype    C.UTF-8</span>
<span class="co">#&gt;  tz       UTC</span>
<span class="co">#&gt;  date     2022-03-03</span>
<span class="co">#&gt;  pandoc   2.7.3 @ /usr/bin/ (via rmarkdown)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package                 </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span>
<span class="co">#&gt;  AnnotationDbi              1.56.2   <span style="color: #555555;">2021-11-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  ape                        5.6-2    <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  assertthat                 0.2.1    <span style="color: #555555;">2019-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  backports                  1.4.1    <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  Biobase                    2.54.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocGenerics               0.40.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocIO                     1.4.0    <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocManager                1.30.16  <span style="color: #555555;">2021-06-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  BiocParallel               1.28.3   <span style="color: #555555;">2021-12-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocSet                  * 1.8.1    <span style="color: #555555;">2021-11-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocStyle                * 2.22.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  Biostrings                 2.62.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  bit                        4.0.4    <span style="color: #555555;">2020-08-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  bit64                      4.0.5    <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  bitops                     1.0-7    <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  blob                       1.2.2    <span style="color: #555555;">2021-07-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  bookdown                   0.24     <span style="color: #555555;">2021-09-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  broom                      0.7.12   <span style="color: #555555;">2022-01-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  bslib                      0.3.1    <span style="color: #555555;">2021-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  cachem                     1.0.6    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  CBEA                     * 0.99.3   <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">local</span></span>
<span class="co">#&gt;  cellranger                 1.1.0    <span style="color: #555555;">2016-07-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  cli                        3.2.0    <span style="color: #555555;">2022-02-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  colorspace                 2.0-3    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  crayon                     1.5.0    <span style="color: #555555;">2022-02-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  DBI                        1.1.2    <span style="color: #555555;">2021-12-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  dbplyr                     2.1.1    <span style="color: #555555;">2021-04-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  DelayedArray               0.20.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  desc                       1.4.0    <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  digest                     0.6.29   <span style="color: #555555;">2021-12-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  dplyr                    * 1.0.8    <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  ellipsis                   0.3.2    <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  evaluate                   0.15     <span style="color: #555555;">2022-02-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  fansi                      1.0.2    <span style="color: #555555;">2022-01-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  fastmap                    1.1.0    <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  fitdistrplus               1.1-6    <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  forcats                  * 0.5.1    <span style="color: #555555;">2021-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  fs                         1.5.2    <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  generics                   0.1.2    <span style="color: #555555;">2022-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  GenomeInfoDb               1.30.1   <span style="color: #555555;">2022-01-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  GenomeInfoDbData           1.2.7    <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  GenomicRanges              1.46.1   <span style="color: #555555;">2021-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  ggplot2                  * 3.3.5    <span style="color: #555555;">2021-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  glue                       1.6.2    <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  goftest                    1.2-3    <span style="color: #555555;">2021-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  gtable                     0.3.0    <span style="color: #555555;">2019-03-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  haven                      2.4.3    <span style="color: #555555;">2021-08-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  hms                        1.1.1    <span style="color: #555555;">2021-09-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  htmltools                  0.5.2    <span style="color: #555555;">2021-08-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  httr                       1.4.2    <span style="color: #555555;">2020-07-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  IRanges                    2.28.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  jquerylib                  0.1.4    <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  jsonlite                   1.8.0    <span style="color: #555555;">2022-02-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  KEGGREST                   1.34.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  kernlab                    0.9-29   <span style="color: #555555;">2019-11-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  knitr                      1.37     <span style="color: #555555;">2021-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  lattice                    0.20-45  <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span>
<span class="co">#&gt;  lazyeval                   0.2.2    <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  lifecycle                  1.0.1    <span style="color: #555555;">2021-09-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  lmom                       2.8      <span style="color: #555555;">2019-03-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  lubridate                  1.8.0    <span style="color: #555555;">2021-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  magrittr                   2.0.2    <span style="color: #555555;">2022-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  MASS                       7.3-55   <span style="color: #555555;">2022-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  Matrix                     1.4-0    <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  MatrixGenerics             1.6.0    <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  matrixStats                0.61.0   <span style="color: #555555;">2021-09-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  memoise                    2.0.1    <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  mixtools                   1.2.0    <span style="color: #555555;">2020-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  modelr                     0.1.8    <span style="color: #555555;">2020-05-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  munsell                    0.5.0    <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  nlme                       3.1-155  <span style="color: #555555;">2022-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  ontologyIndex              2.7      <span style="color: #555555;">2021-02-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  pillar                     1.7.0    <span style="color: #555555;">2022-02-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  pkgconfig                  2.0.3    <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  pkgdown                    2.0.2    <span style="color: #555555;">2022-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  plyr                       1.8.6    <span style="color: #555555;">2020-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  png                        0.1-7    <span style="color: #555555;">2013-12-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  purrr                    * 0.3.4    <span style="color: #555555;">2020-04-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  R6                         2.5.1    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  ragg                       1.2.2    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  Rcpp                       1.0.8    <span style="color: #555555;">2022-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  RCurl                      1.98-1.6 <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  readr                    * 2.1.2    <span style="color: #555555;">2022-01-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  readxl                     1.3.1    <span style="color: #555555;">2019-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  RefManageR               * 1.3.0    <span style="color: #555555;">2020-11-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  reprex                     2.0.1    <span style="color: #555555;">2021-08-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  rlang                      1.0.1    <span style="color: #555555;">2022-02-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  rmarkdown                  2.12     <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  rprojroot                  2.0.2    <span style="color: #555555;">2020-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  RSQLite                    2.2.10   <span style="color: #555555;">2022-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  rstudioapi                 0.13     <span style="color: #555555;">2020-11-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  rvest                      1.0.2    <span style="color: #555555;">2021-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  S4Vectors                  0.32.3   <span style="color: #555555;">2021-11-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  sass                       0.4.0    <span style="color: #555555;">2021-05-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  scales                     1.1.1    <span style="color: #555555;">2020-05-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  segmented                  1.4-0    <span style="color: #555555;">2022-01-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  sessioninfo              * 1.2.2    <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  SingleCellExperiment       1.16.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  stringi                    1.7.6    <span style="color: #555555;">2021-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  stringr                  * 1.4.0    <span style="color: #555555;">2019-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  SummarizedExperiment       1.24.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  survival                   3.3-1    <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.1.2)</span></span>
<span class="co">#&gt;  systemfonts                1.0.4    <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  textshaping                0.3.6    <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  tibble                   * 3.1.6    <span style="color: #555555;">2021-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  tidyr                    * 1.2.0    <span style="color: #555555;">2022-02-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  tidyselect                 1.1.2    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  tidytree                   0.3.8    <span style="color: #555555;">2022-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  tidyverse                * 1.3.1    <span style="color: #555555;">2021-04-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  treeio                     1.18.1   <span style="color: #555555;">2021-11-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  TreeSummarizedExperiment   2.2.0    <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  tzdb                       0.2.0    <span style="color: #555555;">2021-10-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  utf8                       1.2.2    <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  vctrs                      0.3.8    <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  withr                      2.4.3    <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  xfun                       0.30     <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  xml2                       1.3.3    <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  XVector                    0.34.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  yaml                       2.3.5    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  yulab.utils                0.0.4    <span style="color: #555555;">2021-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span>
<span class="co">#&gt;  zlibbioc                   1.40.0   <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #555555;"> [1] /home/runner/work/_temp/Library</span></span>
<span class="co">#&gt; <span style="color: #555555;"> [2] /opt/R/4.1.2/lib/R/library</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.14/BiocStyle" class="external-link">BiocStyle</a></em> (Oleś, 2021) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie, 2021) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em> (Allaire, Xie, McPherson, et al., 2022) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em> (McLean, 2017).</p>
<p>
<a id="bib-allaire2022rmarkdown"></a><a href="#cite-allaire2022rmarkdown">[1]</a><cite> J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.12. 2022. URL: <a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-benaglia2009mixtools"></a><a href="#cite-benaglia2009mixtools">[2]</a><cite> T. Benaglia, D. Chauveau, D. R. Hunter, et al. “mixtools: An R Package for Analyzing Finite Mixture Models”. In: <em>Journal of Statistical Software</em> 32.6 (2009), pp. 1–29. URL: <a href="http://www.jstatsoft.org/v32/i06/" class="external-link">http://www.jstatsoft.org/v32/i06/</a>.</cite>
</p>
<p>
<a id="bib-delignettemuller2015fitdistrplus"></a><a href="#cite-delignettemuller2015fitdistrplus">[3]</a><cite> M. L. Delignette-Muller and C. Dutang. “fitdistrplus: An R Package for Fitting Distributions”. In: <em>Journal of Statistical Software</em> 64.4 (2015), pp. 1–34. URL: <a href="https://www.jstatsoft.org/article/view/v064i04" class="external-link">https://www.jstatsoft.org/article/view/v064i04</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[4]</a><cite> M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX References in R”. In: <em>The Journal of Open Source Software</em> (2017). DOI: <a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-mcmurdie2013phyloseq"></a><a href="#cite-mcmurdie2013phyloseq">[5]</a><cite> P. J. McMurdie and S. Holmes. “phyloseq: An R package for reproducible interactive analysis and graphics of microbiome census data”. In: <em>PLoS ONE</em> 8.4 (2013), p. e61217. URL: <a href="http://dx.plos.org/10.1371/journal.pone.0061217" class="external-link">http://dx.plos.org/10.1371/journal.pone.0061217</a>.</cite>
</p>
<p>
<a id="bib-morgan2021biocparallel"></a><a href="#cite-morgan2021biocparallel">[6]</a><cite> M. Morgan, J. Wang, V. Obenchain, et al. <em>BiocParallel: Bioconductor facilities for parallel evaluation</em>. R package version 1.28.3. 2021. URL: <a href="https://github.com/Bioconductor/BiocParallel" class="external-link">https://github.com/Bioconductor/BiocParallel</a>.</cite>
</p>
<p>
<a id="bib-nguyen2022cbea"></a><a href="#cite-nguyen2022cbea">[7]</a><cite> Q. Nguyen. <em>CBEA: R package for performing CBEA approach</em>. <a href="https://github.com/qpmnguyen/CBEA" class="external-link uri">https://github.com/qpmnguyen/CBEA</a> - R package version 0.99.3. 2022. DOI: <a href="https://doi.org/10.18129/B9.bioc.CBEA" class="external-link">10.18129/B9.bioc.CBEA</a>. URL: <a href="http://www.bioconductor.org/packages/CBEA" class="external-link">http://www.bioconductor.org/packages/CBEA</a>.</cite>
</p>
<p>
<a id="bib-ole2021biocstyle"></a><a href="#cite-ole2021biocstyle">[8]</a><cite> A. Oleś. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.22.0. 2021. URL: <a href="https://github.com/Bioconductor/BiocStyle" class="external-link">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2021language"></a><a href="#cite-2021language">[9]</a><cite> R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2021. URL: <a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-robinson2022broom"></a><a href="#cite-robinson2022broom">[10]</a><cite> D. Robinson, A. Hayes, and S. Couch. <em>broom: Convert Statistical Objects into Tidy Tibbles</em>. <a href="https://broom.tidymodels.org/" class="external-link uri">https://broom.tidymodels.org/</a>, <a href="https://github.com/tidymodels/broom" class="external-link uri">https://github.com/tidymodels/broom</a>. 2022.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[11]</a><cite> H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5–10. URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2019welcome"></a><a href="#cite-wickham2019welcome">[12]</a><cite> H. Wickham, M. Averick, J. Bryan, et al. “Welcome to the tidyverse”. In: <em>Journal of Open Source Software</em> 4.43 (2019), p. 1686. DOI: <a href="https://doi.org/10.21105/joss.01686" class="external-link">10.21105/joss.01686</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[13]</a><cite> H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session Information</em>. <a href="https://github.com/r-lib/sessioninfo#readme" class="external-link uri">https://github.com/r-lib/sessioninfo#readme</a>, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021.</cite>
</p>
<p>
<a id="bib-xie2021knitr"></a><a href="#cite-xie2021knitr">[14]</a><cite> Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report Generation in R</em>. R package version 1.37. 2021. URL: <a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Quang Nguyen.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
